backend/http/http.go

```diff
@@ -601,13 +591,22 @@ func (o *Object) stat(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to stat: %w", err)
 	}
+	return o.stat(ctx, res)
+}
+
+// stat updates info fields in the Object according to HTTP response headers
+func (o *Object) stat(ctx context.Context, res *http.Response) error {
 	t, err := http.ParseTime(res.Header.Get("Last-Modified"))
-	if err != nil {
-		t = timeUnset
+	if err == nil {
+		o.modTime = t
+	}
+
+	size := rest.ParseSizeFromHeaders(res.Header)
```

Comment:
Note that this wants to set the size to `-1` if it isn't known to make rclone's unknown object size handling kick in.